modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -1004,6 +1010,37 @@ else if (objectFieldDBTypes.contains(dbType) &&
 		}
 	}
 
+	private void _validateEncryptedBusinessType(
+			String businessType, long objectDefinitionId)
+		throws PortalException {
+
+		if (!StringUtil.equals(
+				businessType, ObjectFieldConstants.BUSINESS_TYPE_ENCRYPTED)) {
+
+			return;
+		}
+
+		if (Validator.isNull(
+				GetterUtil.getString(
+					PropsUtil.get(PropsKeys.OBJECT_FIELD_ENCRYPTION_SECRET))) ||
+			Validator.isNull(
+				GetterUtil.getString(
+					PropsUtil.get(
+						PropsKeys.OBJECT_FIELD_ENCRYPTION_ALGORITHM)))) {
+
+			throw new NoSuchPropertiesException("Required property is missing");
```

Comment:
@p-albuquerque create an exception and use it in all places we need to check business rules of this business type.

````xml
<exception>ObjectFieldEncrypted</exception>
````

Reply:
divide this props validation in two, one to validate the secret and another to validate the algorithm, messages:

1 - Missing property, PropsKeys.OBJECT_FIELD_ENCRYPTION_ALGORITHM
2 - Missing property, PropsKeys.OBJECT_FIELD_ENCRYPTION_SECRET

Reply:
sort